util/containerutil/shell_shared.go

```diff
@@ -35,6 +60,73 @@ func (sf *shellFrontend) IsAvailable(ctx context.Context) bool {
 	return err == nil
 }
 
+var listRE = regexp.MustCompile(`\s{2,}`)
+
+func (sf *shellFrontend) ContainerList(ctx context.Context) ([]*ContainerInfo, error) {
+	output, err := sf.commandContextOutput(ctx, "ps")
+	if err != nil {
+		return nil, err
+	}
+	ret := []*ContainerInfo{}
+
+	// The Docker & Podman JSON output format differs, so we parse the standard output here.
```

Comment:
:smiling_face_with_tear:

Reply:
https://github.com/earthly/earthly/pull/3420#discussion_r1372141520

Maybe this idea is preferrable?

Reply:
I'm unsure to be honest; just commenting on a lovely constraint you have ended up with; this seems easier than trying to open a PR in podman to fix the json it generates :)